lib/auth/apiserver.go

```diff
@@ -554,10 +554,6 @@ func (s *APIServer) registerUsingToken(auth ClientI, w http.ResponseWriter, r *h
 	if err := httplib.ReadJSON(r, &req); err != nil {
 		return nil, trace.Wrap(err)
 	}
-
-	// Pass along the remote address the request came from to the registration function.
-	req.RemoteAddr = r.RemoteAddr
-
```

Comment:
If I understand correctly this is replaced by 
1. setting `req.RemoteAddr = ""` if client is not a proxy in `(*ServerWithRoles).RegisterUsingToken` and
2. calling `setRemoteAddrFromContext` in if `req.RemoteAddr` is unset in `(*Server).RegisterUsingToken`?

I think that works feels like it's more spread out than it needs to be, could we handle both cases in `*ServerWithRoles`? Or would that be inconvenient for some reason

Reply:
You're right, it will be better, moving into `*ServerWithRoles`